vertica_python/vertica/connection.py

```diff
@@ -566,19 +578,32 @@ def establish_socket_connection(self, address_list):
 
         # Failover: loop to try all addresses
         while addrinfo:
-            (family, socktype, proto, canonname, sockaddr) = addrinfo
+            if not self.proxy:
+                (family, socktype, proto, canonname, sockaddr) = addrinfo
+                # _AddressList filters all addrs to AF_INET and AF_INET6, which both
+                # have host and port as values 0, 1 in the sockaddr tuple.
+                host = sockaddr[0]
+                port = sockaddr[1]
+            else:
+                (family, socktype, proto, canonname, sockaddr, host, port) = addrinfo
             last_exception = None
 
-            # _AddressList filters all addrs to AF_INET and AF_INET6, which both
-            # have host and port as values 0, 1 in the sockaddr tuple.
-            host = sockaddr[0]
-            port = sockaddr[1]
-
             self._logger.info('Establishing connection to host "{0}" on port {1}'.format(host, port))
 
             try:
                 raw_socket = self.create_socket(family)
-                raw_socket.connect(sockaddr)
+                if self.proxy:
+                    self._logger.info(f'Connecting to proxy: {self.proxy}')
+                    proxy_tuple=(self.proxy.split(':')[0], int(self.proxy.split(':')[1]))
+                    raw_socket.connect(proxy_tuple)
+                    fp = raw_socket.makefile(mode='rw')
+                    fp.write('CONNECT %s:%d HTTP/1.0\r\n\r\n' % (host, port))
+                    fp.flush()
+
+                    statusline = fp.readline().rstrip('\r\n')
```

Comment:
HTTP/1.1 200 Connection established if we get a successful connection or different socket error

Reply:
it will. For an example. Just a simulation where proxy look for us to sent HTTP/1.0 or HTTP/1.1. Seems like if we do not specify we send HTTP/1.1. 

Connecting to proxy: proxy.conexus.svc.local:3128
Proxy response is: HTTP/1.0 503 Service Unavailable

Reply:
Proxy response is: HTTP/1.0 503 Service Unavailable
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code